### ISSUE TITLE
求大佬解惑：关于loss和 clm训练

### ISSUE THREAD

**Issue by yuemengrui** [ymcui/Chinese-LLaMA-Alpaca#324]
我在Chinese-llama-13b的词库上增加了中文，然后没有使用peft，直接微调embedding层和lm_head层，发现loss为0：
{'loss': 5.3546, 'learning_rate': 1.6000000000000001e-06, 'epoch': 0.0}                                                       
{'loss': 0.0, 'learning_rate': 3.2000000000000003e-06, 'epoch': 0.0}                                                          
{'loss': 0.0, 'learning_rate': 4.800000000000001e-06, 'epoch': 0.01}                                                          
{'loss': 0.0, 'learning_rate': 6.4000000000000006e-06, 'epoch': 0.01}                                                         
{'loss': 0.0, 'learning_rate': 8.000000000000001e-06, 'epoch': 0.01}                                                          
{'loss': 0.0, 'learning_rate': 9.600000000000001e-06, 'epoch': 0.01}                                                          
{'loss': 0.0, 'learning_rate': 1.1200000000000001e-05, 'epoch': 0.01}  

我本地测试了一下CLM的数据处理，发现input_ids和labels是一模一样的，这个正常吗？我没有接触过CLM训练。我自己觉得input_ids应该和labels是不一样的才对，求大佬帮忙解答一下我的困惑，谢谢🙏。
input_ids:  [36716, 30658, 38046, 36527, 34256, 31824, 33802, 55791, 30785, 32140, 33831, 30470, 32004, 30534, 30214, 31159, 43609, 35380, 32476, 34403, 57240, 31222, 33003, 30470, 29953, 30534, 30214, 35508, 32187, 31616, 30374, 31441]
   labels:  [36716, 30658, 38046, 36527, 34256, 31824, 33802, 55791, 30785, 32140, 33831, 30470, 32004, 30534, 30214, 31159, 43609, 35380, 32476, 34403, 57240, 31222, 33003, 30470, 29953, 30534, 30214, 35508, 32187, 31616, 30374, 31441]
input_ids:  逊前首席科学家韦斯岸的天使投资2006年10月，千橡互动集团收购校内网2008年6月，创建北京乐新创
   labels:  逊前首席科学家韦斯岸的天使投资2006年10月，千橡互动集团收购校内网2008年6月，创建北京乐新创
==========
input_ids:  [32051, 33356, 42194, 56340, 30267, 30505, 30462, 35145, 30333, 37528, 33478, 66612, 30214, 31221, 34477, 32269, 33090, 31398, 34563, 33173, 30940, 30267, 30462, 35145, 30333, 31685, 30214, 89198, 32091, 35430, 32391, 32017]
   labels:  [32051, 33356, 42194, 56340, 30267, 30505, 30462, 35145, 30333, 37528, 33478, 66612, 30214, 31221, 34477, 32269, 33090, 31398, 34563, 33173, 30940, 30267, 30462, 35145, 30333, 31685, 30214, 89198, 32091, 35430, 32391, 32017]
input_ids:  发展尤其是人力资源体系建设。在王慧文发布的内部信中，他提到这是适合交棒的时间点。王慧文称，一直以来都不能很好的处理工作
   labels:  发展尤其是人力资源体系建设。在王慧文发布的内部信中，他提到这是适合交棒的时间点。王慧文称，一直以来都不能很好的处理工作

**Comment by iMountTai** [comment]
因果模型训练，`input_ids`和`labels`是一样的，但是loss为0的情况有多种情况，如依赖版本、torch版本、代码本身的问题等，还请继续调试，欢迎尝试我们的[预训练脚本](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki/%E9%A2%84%E8%AE%AD%E7%BB%83%E8%84%9A%E6%9C%AC)
